### PR TITLE
[Feature] Truly invertible tensordict permutation of dimensions

### LINKED ISSUE #278: [Feature Request] Make `tensordict.permute(2, 1, 0).permute(0, 1, 2)` return the original tensordict
## Motivation

Inverse operations on `Tensordict`s can return the original data.
This is currently not the case for `tensordict.permute`.

## Solution

When calling `tensordict.permute(a, b, c)` we get and instance of a `PermutedTensorDict`.
If the inverse permutation matched `np.argsort([a, b, c])` then the original tensordict should be returned.

One can take inspiration from `Unsqueezed` and `SqueezedTensorDict` classes for this.

We should also cover these in the tests.

### PR DESCRIPTION
**Pull request by ramonmedel**
## Description

Inverse operations on Tensordicts can return the original data. This is currently not the case for tensordict.permute.

We need to return the original tensordict when calling tensordict.permute(a, b, c) we get an instance of a PermutedTensorDict and if we call again the same method with the same parameters in the resulting PermutedTensorDict 

`assert tensordict.permute(dims_list, dim).permute(dims_list, dim) is tensordict`

## Motivation and Context

`close #278`

- [ ] I have raised an issue to propose this change ([required](https://github.com/facebookresearch/rl/blob/main/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/facebookresearch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/test_tensordict.py [hunk 1]
```diff
@@ -296,6 +296,24 @@ def test_permute(device):
     assert torch.sum(t["a"]) == torch.Tensor([0])
 
 
+@pytest.mark.parametrize("device", get_available_devices())
+def test_permute_applied_twice(device):
```
Comment (vmoens) [inline test/test_tensordict.py:300]: It would be cool to move this to `TestTensorDict`
Here's what it will look like

```python
...
class TestTensorDict:
    ...
    def test_permute_applied_twice(self, td_name, device):
        torch.manual_seed(0)
        tensordict = getattr(self, td_name)(device)
        for _ in range(10):
             p =  torch.randperm(4)
             inv_p = p.argsort()
             other_p = inv_p
             while (other_p == inv_p).all():
                 other_p = torch.randperm(4)
             assert tensordict.permute(*p).permute(*inv_p) is tensordict
             assert tensordict.permute(*p).permute(*other_p) is not tensordict
             
```

**Review by vmoens — APPROVED** [review]
Can you use something like the test function i put in the comment instead?
Otherwise it looks great thanks for this!

File: test/test_tensordict.py (hunk 1 above)
Comment (ramonmedel) [reply test/test_tensordict.py:300]: Hi @vmoens , there is no class in the test files, we are using `pytest` and it doesn't work like that

Comment (vmoens) [reply test/test_tensordict.py:300]: Actually it does :-)
Have a look at [this](https://github.com/facebookresearch/rl/blob/850381115c074bba89da23a6a66d7485bbc366af/test/test_tensordict.py#L519)
It would be great if we could fit this test in that. You can pretty much copy paste what i have written above

Comment (ramonmedel) [reply test/test_tensordict.py:300]: Sorry for the confusion @vmoens, I have added your method as you requested

**Review by vmoens — APPROVED** [review]
LGTM thanks for contributing!
